### PR TITLE
PMM-6834 update aws-staging-start to amzn lnx 2

### PR DESCRIPTION
**Pull request by AlyHKafoury**

### CODE REVIEW

File: pmm/aws-staging-start.groovy [hunk 1]
```diff
@@ -304,9 +304,20 @@ pipeline {
...
+                        sudo yum -y install git gcc make automake libtool openssl-devel ncurses-compat-libs
+                        sudo yum -y install svn docker git php php-mysql php-pdo mysql-community-devel mysql-community-client mysql-community-common mysql-community-server
+                        git clone https://github.com/akopytov/sysbench
+                        cd sysbench
+                        ./autogen.sh
+                        ./configure
+                        make
+                        sudo make install
+                        sysbench --version
+                        cd ..
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:317]: why do you need to compile it? Why not just use sysbench from our repo?

File: pmm/aws-staging-start.groovy [hunk 2]
```diff
@@ -304,9 +304,20 @@ pipeline {
...
                         sudo service mysqld start
-                        sudo yum -y install bats --enablerepo=epel
+                        sudo amazon-linux-extras install epel -y
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:319]: By default after this command epel repo is installed but not enabled so for line 320 it is still recommended to use --enablerepo=epel

File: pmm/aws-staging-start.groovy [hunk 3]
```diff
@@ -304,9 +304,20 @@ pipeline {
                         sudo yum -y update --security
                         sudo yum -y install https://repo.percona.com/yum/percona-release-0.1-7.noarch.rpm
                         sudo rpm --import /etc/pki/rpm-gpg/PERCONA-PACKAGING-KEY
-                        sudo yum -y install svn docker sysbench mysql57-server git php php-mysql php-pdo
+                        sudo yum -y install https://dev.mysql.com/get/mysql57-community-release-el7-11.noarch.rpm 
+                        sudo yum -y install git gcc make automake libtool openssl-devel ncurses-compat-libs
+                        sudo yum -y install svn docker git php php-mysql php-pdo mysql-community-devel mysql-community-client mysql-community-common mysql-community-server
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:309]: why do you need to list all mysql-* packages? As I remember they all are dependencies for server package. *devel package is beeded for compilation of sysbench. As it is recommended to use already compiled version of sysbench it is not needed

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: pmm/aws-staging-start.groovy (hunk 3 above)
Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:309]: I wanted to list all the needed packages so if the repo changes we can always get the packages we need, and sysbench is no longer available in repo in linux 2 so I had to compile it from source

File: pmm/aws-staging-start.groovy (hunk 2 above)
Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:319]: it seems the earlier command installed epel and enabled it also I tested it practically

File: pmm/aws-staging-start.groovy (hunk 3 above)
Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:309]: I followed the official amazon linux 2 build for that

https://aws.amazon.com/blogs/database/running-sysbench-on-rds-mysql-rds-mariadb-and-amazon-aurora-mysql-via-ssl-tls/

Comment (atymchuk) [reply pmm/aws-staging-start.groovy:309]: I agree with Evgeniy - we don't need to compile sysbench on every run, `sudo yum -y install sysbench` should just work.

Comment (atymchuk) [inline pmm/aws-staging-start.groovy:309]: git is duplicate, there is already one on line 308

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Comment by atymchuk** [comment]
@AlyHKafoury Does the new linux2 image come from https://aws.amazon.com/amazon-linux-2/release-notes/? I couldn't find it there.

File: pmm/aws-staging-start.groovy (hunk 3 above)
Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:309]: It is not working on amazon linux 2

Comment (AlyHKafoury) [reply pmm/aws-staging-start.groovy:309]: yea will fix that

**Comment by atymchuk** [comment]
The following job can be replayed: https://pmm.cd.percona.com/job/aws-staging-start/27370/console

**Review by atymchuk — APPROVED** [review]

File: pmm/aws-staging-start.groovy (hunk 1 above)
Comment (atymchuk) [reply pmm/aws-staging-start.groovy]: Solved, we now use the one from our repo.

File: pmm/aws-staging-start.groovy (hunk 3 above)
Comment (atymchuk) [reply pmm/aws-staging-start.groovy]: Solved - no devel packaged are installed.

File: pmm/aws-staging-start.groovy [hunk 4]
```diff
@@ -73,7 +73,7 @@ pipeline {
             description: "Query Source for Monitoring",
             name: 'QUERY_SOURCE')
         text(
-            defaultValue: '-e METRICS_RETENTION=192h',
+            defaultValue: '',
```
Comment (puneet0191) [inline pmm/aws-staging-start.groovy:76]: Why is this removed, we use it to verify default docker server behavior when users pass the wrong variable, it should only throw an error in the logs.

Comment (atymchuk) [reply pmm/aws-staging-start.groovy:76]: Sorry, I didn't know that was part of our tests. However, it seams strange and misleading, because I initially took it for the face value and thought that this param is real! I even changed it to 30 days and then saw it wasn't working. 

So, what do we do @puneet0191? Can't we test for illegal env vars in our test pipelines? It seems like tests belong in there.

File: pmm/aws-staging-start.groovy [hunk 5]
```diff
@@ -304,11 +304,14 @@ pipeline {
                         sudo yum -y update --security
                         sudo yum -y install https://repo.percona.com/yum/percona-release-0.1-7.noarch.rpm
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:305]: I would suggest updating this package to our latest version. Currently it is 1.0-25

File: pmm/aws-staging-start.groovy [hunk 6]
```diff
@@ -304,11 +304,14 @@ pipeline {
...
                         sudo rpm --import /etc/pki/rpm-gpg/PERCONA-PACKAGING-KEY
-                        sudo yum -y install svn docker sysbench mysql57-server git php php-mysql php-pdo
-                        sudo service mysqld start
-                        sudo yum -y install bats --enablerepo=epel
+                        sudo yum -y install git svn docker sysbench
+                        sudo yum -y install https://dev.mysql.com/get/mysql57-community-release-el7-11.noarch.rpm 
+                        sudo yum -y install php php-mysqlnd php-pdo mysql-community-client mysql-community-common mysql-community-server
```
Comment (EvgeniyPatlan) [inline pmm/aws-staging-start.groovy:309]:
```suggestion
                        sudo yum -y install php php-mysqlnd php-pdo mysql-community-server
```

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: pmm/aws-staging-start.groovy (hunk 4 above)
Comment (puneet0191) [reply pmm/aws-staging-start.groovy:76]: I understand your point, I will modify the test suite job to handle this, I think we can remove it for now as the default value.

**Review by puneet0191 — APPROVED** [review]

**Comment by puneet0191** [comment]
@atymchuk I discovered an issue with installing the client with providing a specific version, ex. 2.12.0, 2.13.0, it seems https://pmm.cd.percona.com/job/aws-staging-test-job/482/console

**Comment by EvgeniyPatlan** [comment]
> @atymchuk I discovered an issue with installing the client with providing a specific version, ex. 2.12.0, 2.13.0, it seems https://pmm.cd.percona.com/job/aws-staging-test-job/482/console

mm2-client-2.12.0-6.el6.x86_64
It tries to install el6 package while al2 is based on el7

**Comment by puneet0191** [comment]
yeah, I have added a comment on the Jira ticket, probably I will do that changes directly to the branch, after the call with Alex

File: pmm/aws-staging-start.groovy (hunk 6 above)
Comment (atymchuk) [reply pmm/aws-staging-start.groovy]: Thanks!

**Comment by atymchuk** [comment]
> > @atymchuk I discovered an issue with installing the client with providing a specific version, ex. 2.12.0, 2.13.0, it seems https://pmm.cd.percona.com/job/aws-staging-test-job/482/console
> 
> pmm2-client-2.12.0-6.el6.x86_64
> It tries to install el6 package while al2 is based on el7

Fixed!

**Comment by atymchuk** [comment]
Last successful run at https://github.com/Percona-Lab/jenkins-pipelines/pull/725/files

**Review by puneet0191 — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]
